test/exe/main_common_test.cc

```diff
@@ -309,11 +309,13 @@ TEST_P(AdminRequestTest, AdminRequestAfterRun) {
 TEST_P(MainCommonTest, ConstructDestructLogger) {
   VERBOSE_EXPECT_NO_THROW(MainCommon main_common(argc(), argv()));
 
+  // log_msg constructor doesn't initialize all fields, resulting in uninitialized
+  // memory accesses. "static" is a simple way to start from zeroed memory before
+  // the constructor is run, avoiding this problem.
+  // Has been fixed in spdlog; see https://github.com/gabime/spdlog/issues/888.
+  // This change may be reverted after the next spdlog import.
```

Comment:
If it's already fixed, do you want to just update the spdlog SHA? We do that sometimes to pick up fixes for deps?

Reply:
E.g. https://github.com/envoyproxy/envoy/blob/master/bazel/repository_locations.bzl#L103

Reply:
Matt: I took a swing at this, but ended up running into a conflict from importing spdlog at the level the fix occurs at (some commit before that point got rid of msg_log::raw, which we reference).  I could keep tugging on the string, but I think I'm the wrong person for that, since I don't have a sense of spdlog's interface or how we use it.

Would you prefer to land what I have here (and put off the larger update of access to spdlog to the next time we rev the version we use), or for me to hand this issue over to someone who's more familiar with spdlog and its use by envoy?  Maybe @jmarantz ?

Reply:
OK I'm fine either way. @jmarantz do you want to merge this or try to upgrade?

Reply:
I was just poking around our spdlog dependency and I'm not sure I understand it. Looks like we are pulling v1.2.1.tar.gz -- which seems versioned enough. Bu I can't find where in https://github.com/gabime/spdlog it gives me a list of what tarball versions are available. Nor do I understand the relationship between that versioned tarball and the spdlog sha (867a4b7....).

How do you discover what new versioned tarballs are available?

Reply:
Oh, I found https://github.com/gabime/spdlog/releases via @zurcher 's PR. Doesn't look like there's a later version to pull in with the fix. But anyway from reading the spdlog issue, the fix is just to remove the default ctor, so it's not clear we really need anything other than to avoid calling it.

Reply:
Josh, are we reading the same PR?  He added default initializers for several member variables as well.   I'm looking at https://github.com/gabime/spdlog/commit/1b04c222cfc4781fa4e18c890e01465f9515c2dc.

WRT your question about tarballs, it looked to me as if a) you can pull in a tarball pointing at any commit (see Matt's example above; I tried it in spdlog and could get a tarball pointing at the commit containing the fix), and b) the way you get the SHA is by pulling down the tarball and running shasum -a 256 <compressed tarball> on it (which feels like cheating, but I still think is the way to do it).

Reply:
I just issued https://github.com/envoyproxy/envoy/pull/4962 -- simply not calling the 0-arg ctor appears to address the msan issue, without needing to bump the version of spdlog.